src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -571,7 +571,15 @@ export class SearchSource {
           }
         });
       }),
-      map((response) => onResponse(searchRequest, response, options))
+      map((response) => {
+        if (!options.disableShardFailureWarning) {
+          const warnings = options.inspector?.adapter?.getWarnings();
```

Comment:
we should not do this .... this mixes warnings from all requests together ... we should handle this on per request.